### PR TITLE
Convert question-variant library to Typescript

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Changes are restricted to the bare minimum to ensure the renaming retains similarity needed for Git history.

### CODE REVIEW

File: apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js
Comment (nwalters512) [inline apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js:38]: This seems like the kind of thing we could add a `NOT NULL` constraint for, right?

**Review by nwalters512 — APPROVED** [review]

File: apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js
Comment (jonatanschroeder) [reply apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js:38]: Yes, this is not the first time this column has caused me some grief...

File: apps/prairielearn/src/lib/question-variant.ts
Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/question-variant.ts:97]:
```suggestion
 * @param filename
```

File: apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js
Comment (jonatanschroeder) [reply apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js:38]: I will wait until #9257 is merged before merging this one, so that this check can be removed.

**Pull request merged**
